qgreenland/tasks/common/raster.py

```diff
@@ -82,6 +88,38 @@ def run(self):
 
         with temporary_path_dir(self.output()) as tmp_dir:
             tmp_path = os.path.join(tmp_dir, self.filename)
+
+            logger.info(f"Reprojecting {self.layer_cfg['id']}...")
+
+            try:
+                tmp_ds = gdal.Open(ifile, GA_ReadOnly)
+                proj_str = tmp_ds.GetProjection()
+                if not proj_str:
+                    raise Exception('GDAL failed to detect any projection.')
+                # Attempt to do better than the WKT from gdal...
+                try:
+                    proj = pyproj.crs.CRS.from_wkt(proj_str)
+                    proj_str = proj.to_string()
+                except Exception:
+                    pass
+
+                # This is how you close a gdal dataset...
+                del tmp_ds
+
+                logger.info(f'Detected source projection: {proj_str}')
+            except Exception as e:
+                logger.info(f'Failed to detect source projection: {e}')
+
+            if 'srcSRS' in warp_kwargs:
+                logger.info('Overriding the source projection with: '
+                            f"{warp_kwargs['srcSRS']}")
+            elif not proj_str:
```

Comment:
Maybe we should set the `proj_str` to a blank string `''` or `None` prior to the first `try` block...If a new layer is configured that GDAL can't read and hit an exception on opening, `proj_str` will be undefined and an uninformative exception will be raised instead of our `RuntimeError` below.

Reply:
Refactored!